src/wieldmesh.cpp

```diff
@@ -524,36 +572,49 @@ void getItemMesh(Client *client, const ItemStack &item, ItemMesh *result)
 
 		rotateMeshXZby(mesh, -45);
 		rotateMeshYZby(mesh, -30);
-
-		postProcessNodeMesh(mesh, f, false, false, nullptr, &result->buffer_colors);
 	}
 	result->mesh = mesh;
 }
 
 
 
-scene::SMesh *getExtrudedMesh(ITextureSource *tsrc, const std::string &imagename)
+scene::SMesh *getExtrudedMesh(ITextureSource *tsrc,
+	const std::string &imagename, const std::string &overlay_name)
 {
+	// check textures
 	video::ITexture *texture = tsrc->getTextureForMesh(imagename);
 	if (!texture) {
 		return nullptr;
 	}
+	video::ITexture *overlay_texture =
+		(overlay_name == "") ? nullptr : tsrc->getTexture(overlay_name);
```

Comment:
`overlay_name.empty()` and `NULL` here too, as `getTexture` doesn't return `nullptr` on failure (yet).

Reply:
i take a minute to fix it